kotlinx-coroutines-core/common/src/CoroutineScope.kt

```diff
@@ -6,39 +6,34 @@ package kotlinx.coroutines
 
 import kotlinx.coroutines.internal.*
 import kotlinx.coroutines.intrinsics.*
-import kotlin.coroutines.intrinsics.*
 import kotlin.coroutines.*
+import kotlin.coroutines.intrinsics.*
 
 /**
  * Defines a scope for new coroutines. Every coroutine builder
  * is an extension on [CoroutineScope] and inherits its [coroutineContext][CoroutineScope.coroutineContext]
  * to automatically propagate both context elements and cancellation.
  *
+ * The best way to obtain a standalone instance of the scope is [CoroutineScope()] and [MainScope()] factory functions.
+ * Additional context elements can be appended to the scope using [plus][CoroutineScope.plus] operator.
+ *
+ * Manual implementation of this interface is not recommended, implementation by delegation should be preferred instead.
+ * By convention context of the scope should contain an instance of a [job][Job] to enforce structured concurrency.
```

Comment:
"context of the scope" would benefit, IMO, from a hyper-link to `coroutineContext` member value.